src/blockchain/Blockchain.ts

```diff
@@ -47,56 +52,108 @@ export interface IBlockchainApi {
     identity: Identity,
     tx: SubmittableExtrinsic
   ): Promise<SubmittableExtrinsic>
+  submitSignedTx(
+    identity: Identity,
+    tx: SubmittableExtrinsic,
+    opts?: SubscriptionPromiseOptions
+  ): Promise<SubmittableResult>
   submitTx(
     identity: Identity,
     tx: SubmittableExtrinsic,
     opts?: SubscriptionPromiseOptions
   ): Promise<SubmittableResult>
-  getNonce(accountAddress: string): Promise<Codec>
+  getNonce(accountAddress: string): Promise<BN>
+  reSignTx(
+    identity: Identity,
+    tx: SubmittableExtrinsic
+  ): Promise<SubmittableExtrinsic>
 }
+const TxOutdated = '1010: Invalid Transaction: Transaction is outdated'
+const TxPriority = '1014: Priority is too low:'
+const TxAlreadyImported = 'Transaction Already'
 
+export const IS_RELEVANT_ERROR: ErrorEvaluator = (err: Error) => {
+  return new RegExp(
+    `${TxAlreadyImported}|${TxOutdated}|${TxPriority}`,
+    'g'
+  ).test(err.message)
+}
 export const IS_READY: ResultEvaluator = (result) => result.status.isReady
 export const IS_IN_BLOCK: ResultEvaluator = (result) => result.isInBlock
 export const EXTRINSIC_EXECUTED: ResultEvaluator = (result) =>
   ErrorHandler.extrinsicSuccessful(result)
 export const IS_FINALIZED: ResultEvaluator = (result) => result.isFinalized
+export const IS_USURPED: ResultEvaluator = (result) =>
+  result.status.isUsurped && ERROR_TRANSACTION_USURPED()
+export const IS_ERROR: ResultEvaluator = (result) => {
+  return (
+    (result.status.isDropped && Error('isDropped')) ||
+    (result.status.isInvalid && Error('isInvalid')) ||
+    (result.status.isFinalityTimeout && Error('isFinalityTimeout'))
+  )
+}
+export const EXTRINSIC_FAILED: ResultEvaluator = (result) => {
+  return (
+    ErrorHandler.extrinsicFailed(result) &&
+    (ErrorHandler.getExtrinsicError(result) || UNKNOWN_EXTRINSIC_ERROR)
+  )
+}
 
-export const IS_ERROR: ResultEvaluator = (result) =>
-  result.isError && ERROR_UNKNOWN()
-export const EXTRINSIC_FAILED: ResultEvaluator = (result) =>
-  ErrorHandler.extrinsicFailed(result) &&
-  (ErrorHandler.getExtrinsicError(result) || UNKNOWN_EXTRINSIC_ERROR)
+/**
+ * Parses potentially incomplete or undefined options and returns complete [[SubscriptionPromiseOptions]].
+ *
+ * @param opts Potentially undefined or partial [[SubscriptionPromiseOptions]] .
+ * @returns Complete [[SubscriptionPromiseOptions]], with potentially defaulted values.
+ */
+export function parseSubscriptionOptions(
+  opts?: Partial<SubscriptionPromiseOptions>
+): SubscriptionPromiseOptions {
+  const {
+    resolveOn = IS_FINALIZED,
+    rejectOn = (result: SubmittableResult) =>
+      IS_ERROR(result) || EXTRINSIC_FAILED(result) || IS_USURPED(result),
+    timeout,
+  } = { ...opts }
```

Comment:
OMGG YOU CAN SET DEFAULTS LIKE THIS ????? this changes eeeeeverything!

Reply:
but do we need this as an extra function? Whenever I use an options map on a function, I just apply defaults in the first line(s). It's a one-liner and used once - could be integrated.

Reply:
Yeah, it's pretty sexy, innit?
Yes, this could be a one-liner, though this would also mean that passing the whole object referencing these created vars would be ugly IMO.
instead of `parseSubscriptionOptions(opts)` it would be `{resolveOn, rejectOn, timeout }`.
What are your thoughts?

Reply:
I see what you mean... mhh an alternative would be
```js
const defaults = {
     resolveOn: IS_FINALIZED,
     rejectOn: (result: SubmittableResult) =>
       IS_ERROR(result) || EXTRINSIC_FAILED(result) || IS_USURPED(result),
   }
const options = Object.assign(defaults, opts) // unfortunately {...defaults, ...opts} results in weird behaviour
```

Reply:
I ment:

`const defaults = {
     resolveOn: IS_FINALIZED,
     rejectOn: (result: SubmittableResult) =>
       IS_ERROR(result) || EXTRINSIC_FAILED(result) || IS_USURPED(result),
   } = {...opts}

const options = {resolveOn, rejectOn, timeout } `

Reply:
```javascript
const defaults = {
  resolve: IS_FINALIZED,
  rejectOn: (result: SubmittableResult) =>
       IS_ERROR(result) || EXTRINSIC_FAILED(result) || IS_USURPED(result),
}

return {
  ...defaults,
  ...opts
}
```